sdk/resourcemanager/azure-resourcemanager-privatedns/src/main/java/com/azure/resourcemanager/privatedns/models/PrivateDnsZone.java

```diff
@@ -0,0 +1,591 @@
+     * @return the current number of virtual networks that are linked to this Private DNS zone.
+     */
+    long numberOfVirtualNetworkLinks();
+
+    /**
+     * @return the maximum number of virtual networks that can be linked to this Private DNS zone
+     * with registration enabled.
+     */
+    long maxNumberOfVirtualNetworkLinksWithRegistration();
+
+    /**
+     * @return the current number of virtual networks that are linked to this Private DNS zone
+     * with registration enabled.
+     */
+    long numberOfVirtualNetworkLinksWithRegistration();
+
+    /**
+     * @return the provisioning state of the resource.
+     */
+    ProvisioningState provisioningState();
+
+    /** @return the record sets in this zone. */
+    PagedIterable<PrivateDnsRecordSet> listRecordSets();
+
+    /** @return the record sets in this zone asynchronously. */
+    PagedFlux<PrivateDnsRecordSet> listRecordSetsAsync();
+
+    /**
+     * Lists all the record sets in this zone with the given suffix.
+     *
+     * @param recordSetNameSuffix the record set name suffix
+     * @return the record sets
+     */
+    PagedIterable<PrivateDnsRecordSet> listRecordSets(String recordSetNameSuffix);
+
+    /**
+     * Lists all the record sets in this zone with the given suffix asynchronously.
+     *
+     * @param recordSetNameSuffix the record set name suffix
+     * @return the record sets
+     */
+    PagedFlux<PrivateDnsRecordSet> listRecordSetsAsync(String recordSetNameSuffix);
+
+    /**
+     * Lists all the record sets in this zone with each entries in each page limited to the given size.
+     *
+     * @param pageSize the maximum number of record sets in a page
+     * @return the record sets
+     */
+    PagedIterable<PrivateDnsRecordSet> listRecordSets(int pageSize);
+
+    /**
+     * Lists all the record sets in this zone with each entries in each page limited to the given size asynchronously.
+     *
+     * @param pageSize the maximum number of record sets in a page
+     * @return the record sets
+     */
+    PagedFlux<PrivateDnsRecordSet> listRecordSetsAsync(int pageSize);
+
+    /**
+     * Lists all the record sets in this zone with the given suffix, also limits the number of entries per page to the
+     * given page size.
+     *
+     * @param recordSetNameSuffix the record set name suffix
+     * @param pageSize the maximum number of record sets in a page
+     * @return the record sets
+     */
+    PagedIterable<PrivateDnsRecordSet> listRecordSets(String recordSetNameSuffix, int pageSize);
+
+    /**
+     * Lists all the record sets in this zone with the given suffix, also limits the number of entries per page to the
+     * given page size asynchronously.
+     *
+     * @param recordSetNameSuffix the record set name suffix
+     * @param pageSize the maximum number of record sets in a page
+     * @return the record sets
+     */
+    PagedFlux<PrivateDnsRecordSet> listRecordSetsAsync(String recordSetNameSuffix, int pageSize);
```

Comment:
Is is better just provide `PrivateDnsRecordSets<PrivateDnsRecordSet> recordSets()`

Reply:
I think it likely be covered by more specific ones like `aaaaRecordSets`, `aRecordSets`, etc. (so this one is a generic `list` which user not very likely calls)

Reply:
It has `privateZone.aaaaRecordSets().list()` and `privateZone.aRecordSets().list()` while the `listRecordSets()` would list all record sets.